test/whisperkit_test.py

```diff
@@ -0,0 +1,191 @@
+        metadata_file = open(metadata_path)
+        self.metadata = json.load(metadata_file)
+        metadata_text = json.dumps(self.metadata)
+        metadata_file.close()
+
+        if os.path.isdir(self.args.input):
+            for file in os.listdir(self.args.input):
+                full_file_name = os.path.join(self.args.input, file)
+                # checking if it is a file
+                if os.path.isfile(full_file_name) is False:
+                    continue
+                
+                name_only, extension = os.path.splitext(file)
+                if extension not in self.audio_file_ext:
+                    continue
+
+                if metadata_text.find(name_only) == -1:
+                    continue
+
+                self.files.append(file)
+
+    def test_audio_transcription(self):
+        if os.path.exists(OUTPUT_FOLDER) is False:
+            os.mkdir(OUTPUT_FOLDER)
+            
+        with futures.ThreadPoolExecutor(max_workers=len(self.devices)) as executor:
+            future_test = { 
+                executor.submit(self.run_test, device):
+                    device for device in self.devices
+            }
+
+            for future in futures.as_completed(future_test):
+                device = future_test[future]
+                try: 
+                    output_json = future.result()
+                except Exception as exc:
+                    print('device %r generated an exception: %s' % (device, exc))
+                else:
+                    if output_json is None:
+                        print('device %r test has failed..')
+                        self.assertTrue(False)
+                    with open(OUTPUT_FOLDER + device + "_" + TestRunADB.output_file, "w")\
+                        as json_file:
+                        json.dump(output_json, json_file)
+
+
+def download_hg_dataset():
+    test_path = os.path.dirname(os.path.abspath(__file__)) + "/../test/"
+    hf_hub_download(
+                "argmaxinc/whisperkit-test-data",
+                filename="metadata.json",
+                subfolder=TEST_DATASET,
+                local_dir=test_path,
+                repo_type="dataset",
+                revision="main",
+    )
+
+    for file in TEST_FILES:
+        if os.path.exists(test_path + "/" + TEST_DATASET + "/" + file):
+            continue
+        try:
+            hf_hub_download(
+                "argmaxinc/whisperkit-test-data",
+                filename=file,
+                subfolder=TEST_DATASET,
+                local_dir=test_path,
+                repo_type="dataset",
+                revision="main",
+            )
+            print(f'downloaded {file}')
+        except:
+            print(f'can not find {file} from Huggingface')
+        
+    return test_path + "/" + TEST_DATASET
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-m", "--model_size", 
```

Comment:
can you change this to --model-path and have it confirm to the whisperkit model directory structure expectations? 

We're moving towards similar interfaces across the projects, let's bring that in to the testing scripts as well